DOCS-1247-PGD-6.0-Docs (#6763)
EnterpriseDB/docs: product_docs/docs/pgd/6/reference/sequences.mdx

-   `timeshard` &mdash; The older version of SnowflakeId sequence. Provided for
    backward compatibility only. The SnowflakeId is preferred.
-   `distributed` (default) &mdash; A special value that you can use only for
     [`bdr.default_sequence_kind`](/pgd/latest/reference/tables-views-functions/pgd-settings/#global-sequence-parameters). It selects `snowflakeid` for `int8`
     sequences (that is, `bigserial`) and `galloc` sequence for `int4`
     (that is, `serial`) and `int2` sequences.

The [`bdr.sequences`](/pgd/latest/reference/tables-views-functions/catalogs-visible/#bdrsequences) view shows information about individual sequence kinds.

The `currval()` and `lastval()` functions work correctly for all types of global sequences.

## Automatic sequence conversion

> In PGD 6.0 and later, the act of joining a node to a PGD group or creating a new group will also trigger an automatic conversion of any local sequences into global sequences. The [`bdr.default_sequence_kind`](/pgd/latest/reference/pgd-settings/#bdrdefault_sequence_kind) should be set to `distributed`. This will then select the best kind of sequence to convert the local sequences into. If `bdr.default_sequence_kind` is set to `local`, the sequences are left as local sequences. Conversions to `galloc` is performed in a way that ensures that the sequence will not conflict with any other sequences in the group.

Reviewer: ```suggestion
In PGD 6.0 and later, the act of joining a node to a PGD group or creating a new grou also triggers a conversion of any local sequences into global sequences. Set [`bdr.default_sequence_kind`](/pgd/latest/reference/pgd-settings/#bdrdefault_sequence_kind) to `distributed`. This setting then selects the best kind of sequence to convert the local sequences into. If `bdr.default_sequence_kind` is set to `local`, the sequences are left as local sequences. Conversions to `galloc` are performed in a way that ensures that the sequence doesn't conflict with any other sequences in the group.
```

> If you decide to start with local sequences and later switch to galloc sequences, you can do so by setting the `bdr.default_sequence_kind` to `galloc` and then running the `bdr.alter_sequence_set_kind()` function on each sequence you want to convert. Do be aware though that you will need to manually set the starting values of the sequences to ensure that they do not conflict with any existing values in the table. See [converting a local sequence to a galloc sequence](#converting-a-local-sequence-to-a-galloc-sequence) for more information about this in general and specifically [how to set a new start value for a sequence](#2-set-a-new-start-value-for-the-sequence).

Reviewer: ```suggestion
If you decide to start with local sequences and later switch to galloc sequences, you can do so by setting `bdr.default_sequence_kind` to `galloc` and then running the `bdr.alter_sequence_set_kind()` function on each sequence you want to convert. Be aware, though, that you need to manually set the starting values of the sequences to ensure that they don't conflict with any existing values in the table. See [Converting a local sequence to a galloc sequence](#converting-a-local-sequence-to-a-galloc-sequence) for more information about this in general and specifically [How to set a new start value for a sequence](#2-set-a-new-start-value-for-the-sequence).
```

The differences between timeshard and SnowflakeId are as follows:

- Timeshard can generate up to 16384 per millisecond (about 16 million per
  second), which is more than SnowflakeId. However, there's no protection
  against wraparound within a given millisecond. Schemas using the timeshard
  sequence must protect the use of the `UNIQUE` constraint when using timeshard values
  for a given column.
- The timestamp component of timeshard sequence runs out of values in
  the year 2050 and, if used in combination with bigint, the values wrap
  to negative numbers in the year 2033. This means that sequences generated
  after 2033 have negative values. This is a considerably shorter time
  span than SnowflakeId and is the main reason why SnowflakeId is preferred.
- Timeshard sequences require occasional disk writes (similar to standard local
  sequences). SnowflakeIds are calculated in memory so the SnowflakeId
  sequences are in general a little faster than timeshard sequences.

### Unlogged sequences and PGD

Since Postgres 15, it has been possible to create unlogged sequences. These are

> related and similar to unlogged tables, which are not written to the WAL and are not replicated.

Reviewer: ```suggestion
related and similar to unlogged tables, which aren't written to the WAL and aren't replicated.
```

> In the context of PGD and unlogged sequences, it is not a sensible configuration to have an unlogged

Reviewer: ```suggestion
In the context of PGD and unlogged sequences, it isn't a sensible configuration to have an unlogged
```
